src/library/parserm3u.cpp

```diff
@@ -47,84 +47,75 @@ ParserM3u::~ParserM3u()
-    if (file.open(QIODevice::ReadOnly) && !isBinary(sFilename)) {
-        /* Unfortunately, QT 4.7 does not handle <CR> (=\r or asci value 13) line breaks.
-         * This is important on OS X where iTunes, e.g., exports M3U playlists using <CR>
-         * rather that <LF>
-         *
-         * Using QFile::readAll() we obtain the complete content of the playlist as a ByteArray.
-         * We replace any '\r' with '\n' if applicaple
-         * This ensures that playlists from iTunes on OS X can be parsed
-         */
-        QByteArray ba = file.readAll();
-        //detect encoding
-        bool isCRLF_encoded = ba.contains("\r\n");
-        bool isCR_encoded = ba.contains("\r");
-        if(isCR_encoded && !isCRLF_encoded)
-            ba.replace('\r','\n');
-        QTextStream textstream(ba.constData());
-
-        if (isUtf8(ba.constData())) {
-            textstream.setCodec("UTF-8");
-        } else {
-            textstream.setCodec("windows-1252");
-        }
+    // Unfortunately QTextStream does not handle <CR> (=\r or asci value 13) line breaks.
+    // This is important on OS X where iTunes, e.g., exports M3U playlists using <CR>
+    // rather that <LF>.
+    //
+    // Using QFile::readAll() we obtain the complete content of the playlist as a ByteArray.
+    // We replace any '\r' with '\n' if applicaple
+    // This ensures that playlists from iTunes on OS X can be parsed
+    QByteArray ba = file.readAll();
+    //detect encoding
+    bool isCRLF_encoded = ba.contains("\r\n");
+    bool isCR_encoded = ba.contains("\r");
+    if (isCR_encoded && !isCRLF_encoded) {
+        ba.replace('\r', '\n');
+    }
 
-        while (!textstream.atEnd()) {
-            QString sLine = getFilepath(&textstream, basepath);
-            if (sLine.isEmpty()) {
-                continue;
-            }
-            //qDebug() << "ParserM3u: parsed: " << (sLine);
-            m_sLocations.append(sLine);
-        }
+    QTextStream textstream(ba.constData());
+    if (isUtf8(ba.constData())) {
+        textstream.setCodec("UTF-8");
+    } else {
+        textstream.setCodec("windows-1252");
+    }
 
-        file.close();
-        return m_sLocations;
+    const QString basepath = sFilename.section('/', 0, -2);
+    while (!textstream.atEnd()) {
+        QString sLine = getFilepath(&textstream, basepath);
+        if (sLine.isEmpty()) {
+            continue;
+        }
+        m_sLocations.append(sLine);
     }
 
-    file.close();
-    return QList<QString>(); //if we get here something went wrong
+    return m_sLocations;
 }
 
 
 QString ParserM3u::getFilepath(QTextStream* stream, QString basepath) {
-    QString textline = stream->readLine();
-
-    while (!textline.isEmpty()) {
-        //qDebug() << "Untransofrmed text: " << textline;
-        if (textline.isNull()) {
-            break;
+    QString textline;
+    while (!(textline = stream->readLine().trimmed()).isEmpty()) {
+        if (textline.startsWith("#")) {
```

Comment:
Should we allow white space characters in front of ```#```?
Are tailing comments possible?

Reply:
According to the specs this would be I'll formated.
The line will be sorted out later anyway.

Reply:
Reverted back to the old handling and added a comment

Reply:
> Should we allow white space characters in front of `#`?

[Robstness principle](https://en.wikipedia.org/wiki/Robustness_principle)

Ignoring redundant whitespace characters doesn't change the result, let's just do it.